File: waku/v2/node/libwaku/libwaku.c
#include <stdio.h>
#include <stddef.h>

#include "libwaku.h"

int main(int argc, char* argv[]) {
  char* string;
  NimMain();
  //echo();
  string = info("hello there");
  printf("Info: %s", string);
}

File: waku/v2/node/libwaku/libwaku.nim
# libwaku
#
# Exposes a C API that can be used by other environment than C.

# TODO Start a node
# TODO Mock info call
# TODO Write header file
# TODO Write example C code file
# TODO Wrap info call
# TODO Init a node

# proc info*(node: WakuNode): WakuInfo =
proc info(foo: string): string {.exportc.} =
  let s = "info about waku node" & foo
  echo s
  return s

proc echo() {.exportc.} =
 echo "echo"
